### PR TITLE
Cli ui redesign with ansi escapes

### LINKED ISSUE #297: Use console colours and formatting to optimize cortex CLI output
### Description

Currently every title takes 2-3 additional lines. We can take advantage of text formatting like bold and underline to indicate headers and optimize CLI output.

```
...

-----
Title
-----

....
```

### Motivation

CLI output real estate is valuable, as we add more features. We can save lines of output without losing clarity by replacing dashed lines with text formatting in console.

### Additional Context

- https://askubuntu.com/questions/528928/how-to-do-underline-bold-italic-strikethrough-color-background-and-size-i

### PR DESCRIPTION
**Pull request by vishalbollu**
Closes #297 

---
Checklist:
- [X] Run `make test` and `make lint`
- [X] Test end to end manually (e.g. build/push all images, restart operator, and run `cx refresh`)
- [X] Update examples
- [ ] Update documentation (add any new files to `summary.md`)
- [ ] Merge to master
- [ ] Cherry-pick into release branches if it's a bugfix
- [ ] Check [gitbook](https://docs.cortex.dev/v/master/) that docs look good and links function properly
- [ ] Alert team if dev environment changed
- [ ] Delete the branch once it's merged

### CODE REVIEW

File: cli/cmd/get.go [hunk 1]
```diff
@@ -627,7 +627,7 @@ func strMapToStr(strings map[string]string) string {
 	return out
 }
 
-func dataResourceTable(resources []context.Resource, dataStatuses map[string]*resource.DataStatus) string {
+func dataResourceTable(resources []context.Resource, dataStatuses map[string]*resource.DataStatus, title string) string {
```
Comment (deliahu) [inline cli/cmd/get.go]: Maybe `title` should be type `resource.Type`, and in `resource/type.go` we should have a function to get the user-facing type string (and a corresponding map similar to `types` and `typePlurals`?

File: cli/cmd/get.go [hunk 2]
```diff
@@ -478,42 +441,64 @@ func describeAPI(name string, resourcesRes *schema.GetResourcesResponse, flagVer
...
-		out += titleStr("Configuration") + api.UserConfigStr()
+	out := "\n" + console.Bold("url:  ") + apiEndpoint + "\n"
+	out += fmt.Sprintf("%s curl -k -X POST -H \"Content-Type: application/json\" %s -d @samples.json\n\n", console.Bold("curl:"), apiEndpoint)
+	out += fmt.Sprintf(console.Bold("updated at:")+" %s\n", libtime.LocalTimestamp(updatedAt))
+
+	out += "\n"
+	t := table.Table{
+		Headers: headers,
+		Rows: [][]interface{}{
+			row,
```
Comment (deliahu) [inline cli/cmd/get.go]: nit: these three lines can probably just be one line

File: cli/cmd/get.go [hunk 3]
```diff
@@ -665,7 +665,7 @@ func apiResourceTable(apiGroupStatuses map[string]*resource.APIGroupStatus) stri
 		rows = append(rows, []interface{}{
 			name,
 			groupStatus.Available(),
-			groupStatus.UpToDate(),
+			groupStatus.ReadyUpdated,
 			groupStatus.Requested,
 			groupStatus.FailedUpdated,
```
Comment (deliahu) [inline cli/cmd/get.go]: Should we also hide failed here?

File: cli/cmd/get.go [hunk 4]
```diff
@@ -739,7 +736,10 @@ func resourceStatusesStr(resourcesRes *schema.GetResourcesResponse) string {
 	for i, title := range titles {
 		paddingWidth := maxTitleLen - len(title) + 3
 		padding := strings.Repeat(" ", paddingWidth)
-		out += title + ":" + padding + values[i] + "\n"
+		out += title + ":" + padding + values[i]
```
Comment (deliahu) [inline cli/cmd/get.go]: Is there still a new line separating these lines? Was there double new lines before?

File: pkg/lib/table/table.go [hunk 5]
```diff
@@ -116,24 +117,22 @@ func Format(t Table) (string, error) {
 
 	var headerStr string
 	for colNum, header := range t.Headers {
-		headerStr += header.Title
-		if colNum != lastColIndex {
```
Comment (deliahu) [inline pkg/lib/table/table.go]: What is the reason for removing this if?

File: cli/cmd/get.go (hunk 4 above)
Comment (vishalbollu) [reply cli/cmd/get.go]: there was a trailing newline on the last line

**Pull request merged**